modules/fbx/fbx_parser/FBXAnimation.cpp

```diff
@@ -87,18 +87,25 @@ using namespace Util;
 // ------------------------------------------------------------------------------------------------
 AnimationCurve::AnimationCurve(uint64_t id, const ElementPtr element, const std::string &name, const Document & /*doc*/) :
 		Object(id, element, name) {
+	IF_FBX_IS_CORRUPT_RETURN
```

Comment:
There's some tricks in `core/error/error_macros.h` to make it mandatory so that it fails building without it:
```cpp
// (*): See https://stackoverflow.com/questions/257418/do-while-0-what-is-it-good-for

/**
 * If `m_index` is less than 0 or greater than or equal to `m_size`, prints a generic
 * error message and returns from the function. This macro should be preferred to
 * `ERR_FAIL_COND` for bounds checking.
 */
#define ERR_FAIL_INDEX(m_index, m_size)                                                                             \
        do {                                                                                                            \
                if (unlikely((m_index) < 0 || (m_index) >= (m_size))) {                                                     \
                        _err_print_index_error(FUNCTION_STR, __FILE__, __LINE__, m_index, m_size, _STR(m_index), _STR(m_size)); \
                        return;                                                                                                 \
                }                                                                                                           \
        } while (0); // (*)
```

Not necessarily needed to add in your `FBXError.h` as it does make the code ugly, but it helps ensure consistency.